lib/extensions.js

```diff
@@ -0,0 +1,75 @@
+// Imports
+const fs = require('fs')
+const path = require('path')
+const foundationExtensions = require('../app/config').foundationExtensions || ['govuk-frontend']
+
+// Utils
+const pathFromRoot = (...all) => path.join.apply(null, [__dirname, '..'].concat(all))
+const pathToHookFile = packageName => pathFromRoot('node_modules', packageName, 'govuk-prototype-kit.config.json')
+const objectMap = (object, mapFn) => Object.keys(object).reduce((result, key) => {
+  result[key] = mapFn(object[key], key)
+  return result
+}, {})
+const removeDuplicates = arr => arr.filter((i, pos, self) => self.indexOf(i) === pos)
+const filterOutTreversesAndEmpty = part => part && part !== '..'
+const cloneArray = arr => arr.slice(0)
+
+// Reused Functions
+const lookupConfigForPackage = packageName => {
+  if (fs.existsSync(pathToHookFile(packageName))) {
+    return require(pathToHookFile(packageName))
+  } else {
+    return {}
+  }
+}
+
+const getPackageNamesInOrder = () => {
+  const dependencies = require(pathFromRoot('package.json')).dependencies || {}
+  const allNpmDependenciesInAlphabeticalOrder = Object.keys(dependencies).sort()
+  const installedTopPriorityDependencies = foundationExtensions
+    .filter(packageName => allNpmDependenciesInAlphabeticalOrder.indexOf(packageName) > -1)
+
+  return removeDuplicates(installedTopPriorityDependencies.concat(allNpmDependenciesInAlphabeticalOrder))
+}
+
+const getSourceFromItem = item => typeof item === 'string' ? item : item.src
+
+const extensionItemsByType = getPackageNamesInOrder()
+  .reduce((accum, packageName) => Object.assign({}, accum, objectMap(
+    lookupConfigForPackage(packageName),
+    (item, key) => (accum[key] || []).concat(item.map(item => ({packageName, item})))
+  )), {})
+
+const publicUrl = config => config.item.global ? '/assets' : ['', 'extension-assets', config.packageName]
+  .concat(config.item.split('/').filter(filterOutTreversesAndEmpty))
+  .map(encodeURIComponent).join('/')
+
+const fileSystemPath = (config) => pathFromRoot('node_modules', config.packageName, getSourceFromItem(config.item).split('/')
+  .filter(filterOutTreversesAndEmpty)
+  .join(path.sep))
+
+const mappers = {
```

Comment:
do we need the `mappers` object any more? `publicUrlAndFileSystemPath` could come out and all three functions refered to directly in the exports.

Reply:
I think we benefit from it for a couple of reasons.  We export it and it's good to keep those mappers together IMHO.

Reply:
is there a user need to export it?